### ISSUE TITLE
[BUG] Possible sktime pipeline issue (unexpected warning) in time-series classification on panel data

### ISSUE THREAD

**Issue by jmwhyte** [sktime/sktime#5766]
**Describe the bug**
A warning appears if using a sktime `make_pipeline` with `ColumnConcatenator` and `TimeSeriesForestClassifier`.
We do not see this warning if the `make_pipeline` is from sklearn.

**To Reproduce**

```python
import numpy as np
import pandas as pd

# from sktime.pipeline import make_pipeline # uncommenting this line causes the warning to appear.
from sklearn.pipeline import make_pipeline 

from sktime.transformations.panel.compose import ColumnConcatenator
from sktime.classification.interval_based import TimeSeriesForestClassifier

from sktime.datasets import load_basic_motions

Xtrain, ytrain = load_basic_motions("train",return_type="numpy3D")

test_pipe = make_pipeline(ColumnConcatenator(), TimeSeriesForestClassifier())

test_pipe.fit(Xtrain,ytrain)
```

**Expected behavior**
Having observed this at work, I have tried the MWE in the new Binder environment. 
The warning is unexpected, and induces an amount of anxiety ...

"
/opt/conda/lib/python3.11/site-packages/sktime/base/_base_panel.py:240: UserWarning: Data seen by ClassifierPipeline instance has multivariate series, but this ClassifierPipeline instance cannot handle multivariate series. Calls with multivariate series may result in error or unreliable results.
  warn(msg, obj=self)
"

**Additional context**
What does this mean for sktime TSC that is not first converting data to tabular format?

**Versions**
<details>
System:
    python: 3.11.6 | packaged by conda-forge | (main, Oct  3 2023, 10:40:35) [GCC 12.3.0]
executable: /opt/conda/bin/python
   machine: Linux-5.15.0-72-generic-x86_64-with-glibc2.35

Python dependencies:
          pip: 23.3.2
       sktime: 0.25.0
      sklearn: 1.3.1
       skbase: 0.6.2
        numpy: 1.24.4
        scipy: 1.11.3
       pandas: 1.5.3
   matplotlib: 3.8.0
       joblib: 1.3.2
        numba: 0.57.1
  statsmodels: 0.14.0
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
tensorflow_probability: None
</details>

**Comment by achieveordie** [comment]
Hi, thanks for opening an issue.

I did some digging through the code and some related PRs. I can confirm that as long as you know your data and the capability of underlying transformers/estimators, there's _nothing to worry about_. In your case above where you're converting your multivariate data into univariate and then applying a classifier for univariate classification, you have nothing to worry about.

To give you some context:
-  Every Estimator, Composite, etc. have "tags" associated with them that signal the capabilities they possess (eg: can they handle missing values, can they handle multivariate data, etc.).
- For composites, it gets a bit tricky to statically determine the correct set of tags since it is usually based on the underlying elements (consider the case where a preceding transformer can change/impute the data structure to make it conform to the proceeding estimator).
- If we were to assume that we do not want to dynamically change these tags (see PR #2366 for the rationale), there's no way to tell whether the operations that the user is performing are intentional or by mistake.
- Therefore, we provide a warning to make sure a beginner doesn't accidentally create a pipeline where some unintended actions can occur (passing multivariate data to estimators that can only handle univariate data, etc.) but we also do not want to restrict advanced users from creating pipelines that they know are perfectly handling the incoming data (which is why we don't raise a `ValueError` in case such discrepancy is found in a composite.)

I hope this is helpful to you, if you have any follow-up questions please feel free to ask.

**Comment by achieveordie** [comment]
I'd also like to invite @fkiraly, the author of the related code, to this discussion to add or correct the points made above.

**Comment by fkiraly** [comment]
Yes, indeed - the issue is that the tag system is not smart enough to understand that the pipeline should be able to handle multivariate data. It thinks it is able to handle multivariate data if and only if the classifier at the end can.

Similar logic exists for missing data, where tags actually keep track of whether a transformer removes missing data, this is the tag `capability:missing:removes`. A similar tag could be introduced for turning multivariate into univariate, that would turn off the warning.

**Comment by achieveordie** [comment]
Introducing new tags to keep track of such operations makes sense but how are we going to limit the number of tags that could potentially be introduced to make the system 'smart'? Do the original arguments against a more dynamic tag system still stand?

**Comment by jmwhyte** [comment]
Maybe it's a lot easier to put a summary of the explanation given here as part of the warning? That, or put a note somewhere in the docs so that the warning doesn't come as such a surprise?

**Comment by achieveordie** [comment]
It is certainly easier to simply add a small explanation but in an ideal situation we should be able to handle such a scenario without worrying users with too many unnecessary comments/details, a minimalistic approach would work. We'll be able to handle this issue before the next release, if not the least we'll do is add some context to the warning.
